src/Compilers/CSharp/Test/Syntax/Parsing/ExpressionParsingTests.cs

```diff
@@ -3785,5 +3785,281 @@ public void ConditionalExpressionInInterpolation()
+            }
+            EOF();
+        }
+
+        [Fact]
+        public void RangeExpression_Left_WithIndexes()
+        {
+            UsingExpression("^5..");
+            N(SyntaxKind.RangeExpression);
+            {
+                N(SyntaxKind.IndexExpression);
+                {
+                    N(SyntaxKind.CaretToken);
+                    N(SyntaxKind.NumericLiteralExpression);
+                    {
+                        N(SyntaxKind.NumericLiteralToken, "5");
+                    }
+                }
+                N(SyntaxKind.DotDotToken);
+            }
+            EOF();
+        }
+
+        [Fact]
+        public void RangeExpression_NoOperands()
+        {
+            UsingExpression("..");
+            N(SyntaxKind.RangeExpression);
+            {
+                N(SyntaxKind.DotDotToken);
+            }
+            EOF();
+        }
+
+        [Fact]
+        public void RangeExpression_NoOperands_WithOtherOperators()
+        {
+            UsingExpression("1+..<<2");
+            N(SyntaxKind.LeftShiftExpression);
+            {
+                N(SyntaxKind.AddExpression);
+                {
+                    N(SyntaxKind.NumericLiteralExpression);
+                    {
+                        N(SyntaxKind.NumericLiteralToken, "1");
+                    }
+                    N(SyntaxKind.PlusToken);
+                    N(SyntaxKind.RangeExpression);
+                    {
+                        N(SyntaxKind.DotDotToken);
+                    }
+                }
+                N(SyntaxKind.LessThanLessThanToken);
+                N(SyntaxKind.NumericLiteralExpression);
+                {
+                    N(SyntaxKind.NumericLiteralToken, "2");
+                }
+            }
+            EOF();
+        }
+
+        [Fact]
+        public void RangeExpression_DotSpaceDot()
+        {
+            UsingExpression("1. .2",
+                // (1,1): error CS1073: Unexpected token '.2'
+                // 1. .2
+                Diagnostic(ErrorCode.ERR_UnexpectedToken, "1. ").WithArguments(".2").WithLocation(1, 1),
+                // (1,4): error CS1001: Identifier expected
+                // 1. .2
+                Diagnostic(ErrorCode.ERR_IdentifierExpected, ".2").WithLocation(1, 4));
+        }
+
+        [Fact]
+        public void RangeExpression_MethodInvocation_1()
+        {
+            UsingExpression(".. .ToString()",
+                // (1,1): error CS1073: Unexpected token '.'
+                // .. .ToString()
+                Diagnostic(ErrorCode.ERR_UnexpectedToken, "..").WithArguments(".").WithLocation(1, 1));
+        }
+
+        [Fact]
+        public void RangeExpression_MethodInvocation_2()
+        {
+            UsingExpression("1.. .ToString()",
+                // (1,1): error CS1073: Unexpected token '.'
+                // 1.. .ToString()
+                Diagnostic(ErrorCode.ERR_UnexpectedToken, "1..").WithArguments(".").WithLocation(1, 1));
+        }
```

Comment:
>} [](start = 8, length = 1)

Should `.. .ToString()` and `1.. .ToString()` be valid syntax? Consider testing `..2.ToString()` as well.

Reply:
Also: `c?..:b`

---
In reply to: [185244608](https://github.com/dotnet/roslyn/pull/26405#discussion_r185244608) [](ancestors = 185244608)

Reply:
Added tests. I'm not allowing invocation expressions after range operators for now.

---
In reply to: [185257234](https://github.com/dotnet/roslyn/pull/26405#discussion_r185257234) [](ancestors = 185257234,185244608)